bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/components/UIComponentSitemapProvider.java

```diff
@@ -412,50 +427,78 @@ private void addIconColor(EList<ColorArray> iconColor, UIComponent component) {
+                for (Object sourceIcon : (Collection<?>) sourceIcons) {
+                    if (sourceIcon instanceof String) {
+                        String argument = getRuleArgument(sourceIcon.toString());
+                        List<String> conditionsString = getRuleConditions(sourceIcon.toString(), argument);
                         IconRuleImpl iconRule = (IconRuleImpl) SitemapFactory.eINSTANCE.createIconRule();
-                        ConditionImpl condition = (ConditionImpl) SitemapFactory.eINSTANCE.createCondition();
-                        condition.setItem(matcher.group("item"));
-                        condition.setCondition(matcher.group("condition"));
-                        condition.setSign(matcher.group("sign"));
-                        condition.setState(matcher.group("state"));
-                        iconRule.eSet(SitemapPackage.ICON_RULE__CONDITIONS, condition);
-                        iconRule.setArg(matcher.group("arg"));
-                        iconDef.add(iconRule);
-                    } else {
-                        logger.warn("Syntax error in icon rule '{}' for widget {}", sourceIcon, component.getType());
+                        iconRule.eSet(SitemapPackage.ICON_RULE__ARG, argument);
+                        List<ConditionImpl> conditions = getConditions(conditionsString, component, "iconrules");
+                        iconRule.eSet(SitemapPackage.ICON_RULE__CONDITIONS, conditions);
+                        icon.add(iconRule);
                     }
                 }
             }
         }
     }
 
+    private List<ConditionImpl> getConditions(List<String> conditionsString, UIComponent component, String key) {
+        List<ConditionImpl> conditions = new ArrayList<>();
+        for (String conditionString : conditionsString) {
+            Matcher matcher = CONDITION_PATTERN.matcher(conditionString);
+            if (matcher.matches()) {
+                ConditionImpl condition = (ConditionImpl) SitemapFactory.eINSTANCE.createCondition();
+                condition.setItem(matcher.group("item"));
+                condition.setCondition(matcher.group("condition"));
+                condition.setSign(matcher.group("sign"));
+                condition.setState(matcher.group("state"));
+                conditions.add(condition);
+            } else {
+                logger.warn("Syntax error in {} rule condition '{}' for widget {}", key, conditionString,
+                        component.getType());
+            }
+        }
+        return conditions;
+    }
+
+    private String getRuleArgument(String rule) {
+        int argIndex = rule.lastIndexOf("=") + 1;
+        return rule.substring(argIndex).trim();
+    }
+
+    private List<String> getRuleConditions(String rule, @Nullable String argument) {
+        String conditions = rule;
+        if (argument != null) {
+            conditions = rule.substring(0, rule.lastIndexOf(argument)).trim();
+            if (conditions.endsWith("=")) {
```

Comment:
And if there is one or several spaces between = and the "argument" ?

Reply:
The full rule will be something like:
```
Item>=5 = red
```
The passed in argument would be `red`.
If there is an argument (!= null), it is split off and conditions should end with = (after trim). If there is a second = before, it should actually be part of the state string (5 in the example). So I think this is right.

Reply:
I did not take attention that the previous line ended with a call to trim(). So, no problem, you can ignore this comment.